python/src/worker/tasks/list_genes.py

```diff
@@ -6,18 +6,22 @@
 from aws_xray_sdk.core import xray_recorder
 
 from ..config import config
+from ..helpers.r_worker_exception import RWorkerException
 from ..helpers.remove_regex import remove_regex
 from ..result import Result
 from ..tasks import Task
 
 
 class ListGenes(Task):
-    def _format_result(self, result, total):
+    def _format_result(self, result):
+        total = result["full_count"]
+
         # convert result to list of row dicts
-        result = result.to_dict(orient="records")
+        df = pd.DataFrame.from_dict(result["gene_results"])
+        rows = df.to_dict(orient="records")
 
         # Return a list of formatted results.
-        return Result({"total": total, "rows": result})
+        return Result({"total": total, "rows": rows})
```

Comment:
I see some additional changes here apart from the error handling. Can you explain what you are doing here and why?

Reply:
The operations here doesn't seem to compute any additional data, just reformatting the outputs. That's why I moved them under the `_format_result` function. What do you think?